py/objstr.c

```diff
@@ -469,8 +487,15 @@ STATIC mp_obj_t str_join(mp_obj_t self_in, mp_obj_t arg) {
 
     // count required length
     size_t required_len = 0;
+    if (self_type == &mp_type_bytearray) {
+        self_type = &mp_type_bytes;
+    }
     for (size_t i = 0; i < seq_len; i++) {
-        if (mp_obj_get_type(seq_items[i]) != self_type) {
+        const mp_obj_type_t *seq_type = mp_obj_get_type(seq_items[i]);
+        if (seq_type == &mp_type_bytearray) {
+            seq_type = &mp_type_bytes;
+        }
+        if (seq_type != self_type) {
```

Comment:
I think you're missing a feature gate here? Causing the minimal build to fail.
~~~
#ifdef MICROPY_PY_BUILTINS_BYTEARRAY
~~~

Reply:
Oh I thought I'd fixed that already... might have force pushed away that fix. I'll certainly need to fix that